File: frontend/src/api/portfolio.ts
import apiClient from './client';

export interface PortfolioItem {
  id: string;
  mua_id: string;
  title: string;
  description?: string;
  image_url: string;
  service_type?: string;
  created_at: string;
}

export interface CreatePortfolioRequest {
  title: string;
  description?: string;
  image_url: string;
  service_type?: string;
}

export interface UpdatePortfolioRequest {
  title?: string;
  description?: string;
  image_url?: string;
  service_type?: string;
}

export const portfolioApi = {
  getPortfolioItems: async (): Promise<PortfolioItem[]> => {
    const response = await apiClient.get('/portfolio');
    return response.data;
  },

  createPortfolioItem: async (data: CreatePortfolioRequest): Promise<PortfolioItem> => {
    const response = await apiClient.post('/portfolio', data);
    return response.data;
  },

  updatePortfolioItem: async (id: string, data: UpdatePortfolioRequest): Promise<PortfolioItem> => {
    const response = await apiClient.put(`/portfolio/${id}`, data);
    return response.data;
  },

  deletePortfolioItem: async (id: string): Promise<{ message: string }> => {
    const response = await apiClient.delete(`/portfolio/${id}`);
    return response.data;
  },
};

File: backend/src/models/portfolio.rs
use serde::{Deserialize, Serialize};
use sqlx::FromRow;
use uuid::Uuid;
use chrono::{DateTime, Utc};

#[derive(Debug, Serialize, Deserialize, FromRow)]
pub struct PortfolioItem {
    pub id: Uuid,
    pub mua_id: Uuid,
    pub title: String,
    pub description: Option<String>,
    pub image_url: String,
    pub service_type: Option<String>,
    pub created_at: DateTime<Utc>,
}

#[derive(Debug, Deserialize)]
pub struct CreatePortfolioRequest {
    pub title: String,
    pub description: Option<String>,
    pub image_url: String,
    pub service_type: Option<String>,
}